.github/workflows/release.yml

```
@@ -7,103 +8,63 @@ defaults:
-          "${{ inputs.release_type }}" `
-          "v${{ steps.get-version.outputs.version }}" `
-          "${{ env.project-name }}";
-
-    - name: Validate Release Notes
-      run: |
-        deno run `
-          --allow-read `
-          https://raw.githubusercontent.com/KinsonDigital/Infrastructure/v3.0.3/cicd/validate-notes.ts `
-          "${{ inputs.release_type }}" `
-          "v${{ steps.get-version.outputs.version }}";
-          
-    - name: Preview Release
-      uses: softprops/action-gh-release@v1
-      if: inputs.release_type == 'Preview'
-      with: 
-        name: "🚀Preview Release - v${{ steps.get-version.outputs.version }}" 
-        body_path: "${{ github.workspace }}/Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v${{ steps.get-version.outputs.version }}.md"
-        files: "${{ github.workspace }}/Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v${{ steps.get-version.outputs.version }}.md"
-        tag_name: "v${{ steps.get-version.outputs.version }}"
-        prerelease: true
-        draft: false
-
-    - name: Production Release
-      uses: softprops/action-gh-release@v1
-      if: inputs.release_type == 'Production'
-      with:
-        name: "🚀Production Release - v${{ steps.get-version.outputs.version }}" 
-        body_path: "${{ github.workspace }}/Documentation/ReleaseNotes/ProductionReleases/Release-Notes-v${{ steps.get-version.outputs.version }}.md"
-        files: "${{ github.workspace }}/Documentation/ReleaseNotes/ProductionReleases/Release-Notes-v${{ steps.get-version.outputs.version }}.md"
-        tag_name: "v${{ steps.get-version.outputs.version }}"
-        prerelease: false
-        draft: false
+      - name: Get Build Config
+        id: get-build-config
+        run: |
+          # If the branch that the workflow is running on is the required branch for the release, do a release build
+          if ( "${{ github.base_ref }}" -eq "main") {
+            "build-config=Release" >> $env:GITHUB_OUTPUT;
+            echo "Release build created.";
+          } else { # Any other branch than the release branch, do a debug build
+            "build-config=Debug" >> $env:GITHUB_OUTPUT;
+            echo "Debug build created.";
+          }
+
+
+  perform_release:
+    name: Performing ${{ inputs.release_type }} Release of ${{ vars.PROJECT_NAME }} (${{ needs.determine_build_config.outputs.build-config }})
+    needs: determine_build_config
+    uses: KinsonDigital/Infrastructure/.github/workflows/dotnet-action-release.yml@v8.1.0
+    with:
+      project-name: "${{ vars.PROJECT_NAME }}"
+      release-type: "${{ inputs.release_type }}"
+      run-branch: "${{ github.ref_name }}"
+      net-sdk-version: "${{ vars.NET_SDK_VERSION }}"
+      relative-release-notes-dir-path: "${{ vars.SCRIPT_RELATIVE_DIR_PATH }}"
+      release-notes-file-name-prefix: "${{ vars.RELEASE_NOTES_FILE_NAME_PREFIX }}"
+      build-config: "${{ needs.determine_build_config.outputs.build-config }}"
```

Change this workflow input to:

```suggestion
      build-config: ${{ inputs.release-type == 'Production' && 'Release' || 'Debug' }}
```